src/main/java/teammates/sqllogic/core/FeedbackResponsesLogic.java

```diff
@@ -79,11 +79,29 @@ public boolean isResponseOfFeedbackQuestionVisibleToInstructor(FeedbackQuestion
         return question.isResponseVisibleTo(FeedbackParticipantType.INSTRUCTORS);
     }
 
+    /**
+     * Checks whether a giver has responded a session.
+     */
+    public boolean hasGiverRespondedForSession(String giverIdentifier, List<FeedbackQuestion> questions) {
+        assert questions != null;
+
+        for (FeedbackQuestion question : questions) {
+            boolean hasResponse = question
+                    .getFeedbackResponses()
+                    .stream()
+                    .anyMatch(response -> response.getGiver().equals(giverIdentifier));
+            if (hasResponse) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
     /**
      * Creates a feedback response.
      * @return the created response
      * @throws InvalidParametersException if the response is not valid
-     * @throws EntityAlreadyExistsException if the response already exist
      */
     public FeedbackResponse createFeedbackResponse(FeedbackResponse feedbackResponse)
             throws InvalidParametersException, EntityAlreadyExistsException {
```

Comment:
Exception in declared in the method signature, any reason for removing this?

Reply:
Sorry I think that's my mistake, will change it back.